### PR TITLE
Do 1061 intellinote

### PR DESCRIPTION
**Pull request by dominiksteiner**
@joncodo @amirmehmood please review

### CODE REVIEW

File: app/directives/datePicker/datePicker.directive.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
+var DatePicker = function($document) {
```
Comment (joncodo) [inline app/directives/datePicker/datePicker.directive.js]: I'm surprised you did not use an angular date picker for this. Was there a reason to go custom?

File: app/directives/list/list.directive.js [hunk 2]
```diff
@@ -1,33 +1,90 @@
...
+      items: '=?',
+      value: '=?',
+      visible: '=',
+      parent: '=',
+      searchVisible: '='
+    },
+    replace: true,
+    template: require("./list.template.jade"),
+    link: function(scope, element, attrs) {
+      // var listItemsEl = element[0].querySelector('.list-items');
```
Comment (joncodo) [inline app/directives/list/list.directive.js]: is this work yet to be done? Or can this be removed

File: app/directives/list/list.template.jade [hunk 3]
```diff
@@ -1,8 +1,14 @@
 span.list
 	span.value(ng-click="visible = !visible")
-		img.icon(ng-if="value && value.icon || init && init.icon", src='{{value.icon}}')
-		a.label(ng-if="(value && value.initials || value.label) || (init && init.initials || init.label)", ng-bind-html='value.initials || value.label || init.initials || init.label')
-	.list-items(ng-show='visible')
-		.list-item(ng-repeat='item in items track by $index', ng-click='handleClick(item)')
-			img.icon(ng-if="item.icon", src='{{item.icon}}')
-			a.label(ng-if="item.label", ng-bind-html='item.label')
\ No newline at end of file
+		img.icon(ng-if="value && value.icon || value && value.iconClass", src='{{value.icon}}', class='icon-{{value.iconClass}}')
```
Comment (joncodo) [inline app/directives/list/list.template.jade]: can I get a screenshot of this directive in action?

File: app/directives/spin/spin.directive.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+var Spin = function() {
```
Comment (joncodo) [inline app/directives/spin/spin.directive.js]: we already have a directive for this in hub-lib. You should re-use it

Comment (joncodo) [inline app/directives/spin/spin.directive.js]: I think its called spinner

File: app/directives/datePicker/datePicker.directive.js (hunk 1 above)
Comment (dominiksteiner) [inline app/directives/datePicker/datePicker.directive.js]: @szafarcin can you comment why we didn't use angular date picker?

File: app/styles/dropDownMenu.styl [hunk 5]
```diff
@@ -0,0 +1,147 @@
+@import 'colors'
```
Comment (joncodo) [inline app/styles/dropDownMenu.styl]: there are also angular drop down directives. Maybe that would work here as well?

File: app/directives/list/list.directive.js (hunk 2 above)
Comment (dominiksteiner) [inline app/directives/list/list.directive.js]: just some refresh the list if we scroll to the bottom functionality

File: app/directives/datePicker/datePicker.directive.js (hunk 1 above)
Comment (joncodo) [inline app/directives/datePicker/datePicker.directive.js]: did @szafarcin work on this too? I don't see him in any commits

File: app/directives/spin/spin.directive.js (hunk 4 above)
Comment (dominiksteiner) [inline app/directives/spin/spin.directive.js]: yes, helping you with https://github.com/BroadsoftLabs/hubCore/issues/675 here and making the size configurable, we should deprecate then the hubLib one

File: app/directives/datePicker/datePicker.directive.js (hunk 1 above)
Comment (dominiksteiner) [inline app/directives/datePicker/datePicker.directive.js]: he committed it first in hubLib, i moved it over to hubUi

Comment (joncodo) [inline app/directives/datePicker/datePicker.directive.js]: ah. Ok. Makes sense to use a directive that was made by a bigger team then. So we don't have to maintain it and we get all the features of a bigger project. Also, their directive is very tested. this is not.

There are many of these online. Angular-material project is the nicest but its a big suite of tools. Makes a much more modern UI. For a stand-alone one, you could try https://github.com/g00fy-/angular-datepicker

File: app/directives/list/list.template.jade (hunk 3 above)
Comment (dominiksteiner) [inline app/directives/list/list.template.jade]: screen shot 2016-07-19 at 08 21 39

File: app/directives/datePicker/datePicker.directive.js (hunk 1 above)
Comment (dominiksteiner) [inline app/directives/datePicker/datePicker.directive.js]: @jbaird2000 @dfaberLTM please assess which calendar version we want to use, right now we customised our own which looks like this

screen shot 2016-07-19 at 08 26 40

Benefits : full control of source code and styles
Drawback : we have to maintain it

This is the calendar suggested by @joncodo 

screen shot 2016-07-19 at 08 26 32

Benefits : we don't have to maintain it
Drawback : limit configuration possibilities, would need to adapt our source code to it, might prove to load more assets while ours is all bundled in web pack

Comment (szafarcin) [inline app/directives/datePicker/datePicker.directive.js]: @dominiksteiner @joncodo We are also using open-source angular directive: https://github.com/jkuri/ngDatepicker . The main reason of picking it was easy to modify templates and styles (custom ones going with Hub UI) and we can add more features to it, one I had in mind was adding ability to select hours/minutes as well while selecting timelines (due dates) or expanding it to look like intellinote web date selector in future if required (they are using jquery based datepicker which is much fancier)

**Comment by dfaberLTM** [comment]
@dominiksteiner is your question about the calendar style above still relevant or did you guys already decide on something.  I don't follow the conversation after you asked the question. @jbaird2000

**Comment by dominiksteiner** [comment]
@dfaberLTM we have to do some restructuring work to not include the https://github.com/jkuri/ngDatepicker directly in our repos but fork it, so unless you have something to note on the UI of above first screenshot of the calendar then we should be fine.

**Comment by dominiksteiner** [comment]
@joncodo pointed to the ng-Datepicker fork in our repos now, please review again

**Comment by joncodo** [comment]
lgtm

**Pull request merged**